githubapi-frontend/src/app/shared/service/service.service.ts

```diff
@@ -27,12 +26,15 @@ export class ServiceService {
     return this.http.post<Response>(`${this.API}/${name}`, body);
   }
 
-  list() {
-    return this.http.get(this.API).pipe(tap())
+  list(dado: string) {
+    return this.http.get<SearchResultUser>(`${this.API}users/${dado}`)
   }
 
-  adicionar(dado:string): void{
+  getUser(dado: string): void {
```

Comment:
Avaliar a usabilidade deste método, Está correta a implementação dele?
Não deveria utilizar apenas um sinal de igual ? `=`

Segue o exemplo: 
```ts
this.user = res;
```

Reply:
não entendi

Reply:
Da uma olhada no método `getUser`. Não tem um retorno e vc esta apenas verificando se o `this.users` é igual a resposta.

```ts
getUser (dado: string): void {
  this.http.get(`${this.API}/${dado}`).subscribe(res => this.users === res)
}
```

Um dos problemas é, o que esse método vai fazer exatamente? Se for para obter apenas um usuário, a nomenclatura está correta porém a implementação não. 
Segue o exemplo abaixo.
```ts
getUser (dado: string): Observable<User> {
  return this.http.get<User>(`${this.API}/${dado}`);
}
```
O método agora ele obtem um usuário da api do github com o parametro que vc passar pra ele (nome do usuário). 
Agora ele retorna um `Observable<User>`. Implementando dessa forma vc pode fazer a chamada no seu componente dessa forma:

```ts
...
  ngOnInit() {
    this.service.getUser("eduardoaraya")
       .subscribe(res => this.user = res)
  }
```

Reply:
não chamei dentro ngOnInit, chamei direto dentro da função searchUser no component Home